xapian-core/tests/api_sorting.cc

```diff
@@ -321,79 +321,77 @@ DEFINE_TESTCASE(sortfunctorremote1, remote) {
 }
 
 DEFINE_TESTCASE(replace_weights, backend) {
-    // open the database (in this case a simple text file
-    // we prepared earlier)
 
     Xapian::Database mydb(get_database("apitest_onedoc"));
 
     Xapian::Enquire enquire(mydb);
 
-    // make a simple query, with one word in it - "word".
     enquire.set_query(Xapian::Query("word"));
 
-    // retrieve the top ten results (we only expect one)
+    // retrieve the top ten results (we only expect one).
     Xapian::MSet mymset = enquire.get_mset(0, 10);
 
-    // assigning new weight of -1 to the document.
+    /* assigning new weight of -1 to the document.
+     * a negative weight was chose to avoid matching the positive weight returned by enquire.
+     */
     vector<double> weights {-1};
     mymset.replace_weights(weights.begin(), weights.end());
     Xapian::MSetIterator i = mymset.begin();
-    // checking wether the weight was assigned correctly
+    // The weight for the document should be updated to double(-1).
     TEST_EQUAL_DOUBLE(i.get_weight(), double(-1));
-    // checking whether the max_attained attribute was updated correctly
+    // max_attained for the weight of the single document {double(-1)} = double(-1).
     TEST_EQUAL_DOUBLE(mymset.get_max_attained(), double(-1));
+    // max_possible should be max(max_possible of the previous weights, max_attained for the new weights).
+    TEST_EQUAL_DOUBLE(mymset.get_max_possible(), max(mymset.get_max_attained(), mymset.get_max_possible()));
 
     return true;
 }
 
 DEFINE_TESTCASE(replace_weights2, backend) {
-    // open the database (in this case a simple text file
-    // we prepared earlier)
 
     Xapian::Database mydb(get_database("apitest_onedoc"));
 
     Xapian::Enquire enquire(mydb);
 
-    // make a simple query, with one word in it - "word".
     enquire.set_query(Xapian::Query("word"));
 
-    // retrieve the top ten results (we only expect one)
+    // retrieve the top ten results (we only expect one).
     Xapian::MSet mymset = enquire.get_mset(0, 10);
 
-    // Checking for exception when the number of weights in the input exceeds the number of documents
+    // Checking for exception when the number of weights in the input exceeds the number of documents.
```

Comment:
Again, none of these comments, including this one, says anything beyond the code it comments.